tokio-threadpool/src/task/mod.rs

```diff
@@ -77,13 +87,36 @@ impl Task {
 
         Task {
             state: AtomicUsize::new(State::stub().into()),
+            home_worker: AtomicUsize::new(NONE_WORKER_ID),
             blocking: AtomicUsize::new(BlockingState::new().into()),
             next: AtomicPtr::new(ptr::null_mut()),
             next_blocking: AtomicPtr::new(ptr::null_mut()),
             future: UnsafeCell::new(Some(task_fut)),
         }
     }
 
+    /// Try initializing this task's home worker.
+    ///
+    /// Returns `true` on success, or `false` if the home worker was already initialized.
+    pub fn init_home_worker(&self, id: &WorkerId) -> bool {
+        if self.home_worker.load(Acquire) == NONE_WORKER_ID {
```

Comment:
This can almost certainly be `Relaxed` as the only access to `home_worker` is done when the caller holds the "scheduled" task lock. i.e. there can be no concurrent access to this field (now).

That said, It isn't a big deal to to keep as is.

Reply:
You're right 👍